aocr/util/data_augmentation.py

```diff
@@ -0,0 +1,199 @@
+import random
+from typing import Tuple
+
+import cv2
+import numpy as np
+from PIL import Image, ImageEnhance, ImageOps
+
+
+__all__ = ['add_random_lines',
+           'add_random_padding',
+           'crop_image',
+           'modify_brightness',
+           'modify_contrast',
+           'random_resize',
+           'random_rotation',
+           'modify_sharpness'
+           ]
+
+
+def add_random_lines(img: Image.Image, num_lines_range: Tuple[int, int] = (1, 3)) -> Image.Image:
+    '''
+    Add random lines mimicking scanning artifacts.
+
+    Args:
+        img: Image to modify
+        num_lines_range: Minimum and maximum number of lines to be added
+
+    Returns:
+        A modified Pillow Image.
+    '''
+    img = np.array(img)
+    H, W = img.shape[:2]
+
+    num_lines = random.randint(*num_lines_range)
+    probability = random.random()
+
+    # Only 50% of the images will have lines added
+    if probability > 0.5:
+
+        for _ in range(num_lines):
+            xmin = random.randint(1, W)
+            ymin = random.randint(1, H)
+            xmax = random.randint(xmin, W)
+            ymax = random.randint(ymin, H)
+
+            probability = random.random()
+
+            if probability > 0.95:
+                thickness = 1
+                color = [255] * 3
+            elif probability > 0.9:
+                thickness = 1
+                color = [0] * 3
+            elif probability > 0.7:
+                thickness = 1
+                color = [random.randint(1, 255)] * 3
+            else:
+                thickness = random.randint(1, 2)
+                color = int(np.bincount(img.flatten()).argmax())
```

Comment:
Minor optimization possible here
```suggestion
            thickness = 1
            if probability > 0.95:
                color = [255] * 3
            elif probability > 0.9:
                color = [0] * 3
            elif probability > 0.7:
                color = [random.randint(1, 255)] * 3
            else:
                thickness = random.randint(1, 2)
                color = int(np.bincount(img.flatten()).argmax())
```